src/index.js

```diff
@@ -15,7 +15,10 @@ export default function promiseMiddleware({ dispatch }) {
     return isPromise(action.payload)
       ? action.payload.then(
           result => dispatch({ ...action, payload: result }),
-          error => dispatch({ ...action, payload: error, error: true })
+          error => {
+            dispatch({ ...action, payload: error, error: true });
+            return Promise.reject(error);
+          }
```

Comment:
I think there's one simpler way that is returning the original promise (`action.payload`) instead of returning the result of `action.payload.then()`